From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Sat, 19 Oct 2013 12:14:35 +0300
Subject: [PATCH 107/223] RDMA/cxgb4: Fix four byte info leak in
 c4iw_create_cq()
Patch-mainline: v3.15-rc1
Git-commit: e24a72a3302a638d4c6e77f0b40c45cc61c3f089
References: bsc#909577 FATE#317550

There is a four byte hole at the end of the "uresp" struct after the
->qid_mask member.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Acked-by: David Chang <dchang@suse.com>
Signed-off-by: John Jolly <jjolly@suse.de>
---
 drivers/infiniband/hw/cxgb4/cq.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/infiniband/hw/cxgb4/cq.c
+++ b/drivers/infiniband/hw/cxgb4/cq.c
@@ -930,6 +930,7 @@ struct ib_cq *c4iw_create_cq(struct ib_d
 		if (!mm2)
 			goto err4;
 
+		memset(&uresp, 0, sizeof(uresp));
 		uresp.qid_mask = rhp->rdev.cqmask;
 		uresp.cqid = chp->cq.cqid;
 		uresp.size = chp->cq.size;
